libsql-client/src/lib.rs

```diff
@@ -191,8 +199,19 @@ impl Connection {
     ) -> Self {
         let username = username.into();
         let pass = pass.into();
+        let url = url.into();
+        // Auto-update the URL to start with https://
+        let url = if url.starts_with("https://") {
+            url
+        } else if let Some(stripped) = url.strip_prefix("http://") {
+            "https://".to_owned() + stripped
```

Comment:
Do we really want to silently switch to `https` even if the user explicitly specified `http`? There are valid uses for HTTP (for example for local testing), and it might be extremely confusing for the user if we just unconditionally replace `http` with `https`.

Reply:
Fair point, I'll amend the code so that only protocolless strings are upgraded to https